app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,36 @@
+class MarketsController < ApplicationController
+  def index
+    @markets = Market.all
+  end
+
+  def new
+    @market = Market.new
+  end
+
+  def create
+    Market.create(market_params)
+    redirect_to markets_path
+  end
+
+  def edit
+    @market = Market.find(params[:id])
```

Comment:
Because we use `Market.find(params[:id])` in a few different methods, we could DRY this out slightly by moving that line into a `before_action` line.